lawlibrary/views/court.py

```diff
@@ -0,0 +1,66 @@
+from django.shortcuts import get_object_or_404
+
+from africanlii.utils import lowercase_alphabet
+from peachjam.models import Author, Judgment
+from peachjam.views.generic_views import FilteredDocumentListView
+
+
+class BaseCourtDetailView(FilteredDocumentListView):
+    """Generic List View class for filtering a court's judgments."""
+
+    context_object_name = "documents"
+    paginate_by = 20
+    model = Judgment
+    template_name = "lawlibrary/court_detail.html"
+
+    def get_base_queryset(self):
+        if "year" in self.kwargs:
+            return self.model.objects.filter(
+                author=self.author, date__year=self.kwargs["year"]
+            )
+        return self.model.objects.filter(author=self.author)
```

Comment:
You're repeating yourself here. Rather, get the basic filtered queryset based on the author, and then do the year filtering on top of that if a year is present:

```python
qs = self.model.objects.filter(author=self.author)
if "year" in self.kwargs:
    qs = qs.filter(date__year=self.kwargs["year"])
```